static/app/views/performance/newTraceDetails/traceModels/traceTree.tsx

```diff
@@ -218,6 +218,42 @@ function isJavascriptSDKTransaction(transaction: TraceTree.Transaction): boolean
   );
 }
 
+function isPageloadTransaction(node: TraceTreeNode<TraceTree.NodeValue>): boolean {
+  return isTransactionNode(node) && node.value['transaction.op'] === 'pageload';
+}
+
+function isSSRTransaction(node: TraceTreeNode<TraceTree.NodeValue>): boolean {
+  return isTransactionNode(node) && node.value['transaction.op'] === 'http.server';
+}
+
+function isBrowserRequestSpan(value: TraceTree.Span): boolean {
+  return value.op === 'browser' && value.description === 'request';
+}
+
+/**
+ * Swaps the two nodes in the graph.
+ */
+function childParentSwap({
+  parent,
+  child,
+}: {
+  child: TraceTreeNode<TraceTree.NodeValue>;
+  parent: TraceTreeNode<TraceTree.NodeValue>;
+}) {
+  const parentOfParent = parent.parent!;
+
+  parent.invalidate(parent);
+  child.invalidate(child);
```

Comment:
Dumb question: Wouldn't it make more sense to invalidate _after_ the mutation?

Reply:
I've moved this out of the swap fn and added a comment. The only reason we need it is because we compute and cache some properties (like the depth of each node) and because reparenting might happen due to incremental trace loading (like in replays), we need to invalidate those fields.